lib/fog/storage/google_json/models/file.rb

```diff
@@ -45,7 +45,16 @@ def predefined_acl=(new_predefined_acl)
         end
 
         def body
-          last_modified && (file = collection.get(identity)) ? attributes[:body] ||= file.body : attributes[:body] ||= ""
+          return attributes[:body] if attributes.key?(:body)
+
+          file = collection.get(identity)
+
+          attributes[:body] =
+            if file
+              file.attributes[:body]
```

Comment:
I opted to do this and ditch `last_modified` to avoid any potential of an infinite recursion. I'm not quite sure what `last_modified` has to do with this in any case.